internal/httptransport/httptransport.go

```diff
@@ -132,6 +132,25 @@ type bodyWrapper struct {
 	tid int64
 }
 
+func (bw *bodyWrapper) Read(b []byte) (n int, err error) {
+	start := time.Now()
+	n, err = bw.ReadCloser.Read(b)
+	stop := time.Now()
+	bw.t.Handler.OnMeasurement(model.Measurement{
+		HTTPResponseBodyPart: &model.HTTPResponseBodyPartEvent{
+			// "Read reads up to len(p) bytes into p. It returns the number of
+			// bytes read (0 <= n <= len(p)) and any error encountered."
+			Data:          b[:n],
+			Duration:      stop.Sub(start),
+			Error:         err,
+			NumBytes:      int64(n),
+			Time:          stop.Sub(bw.t.Beginning),
```

Comment:
This also looks like a Duration, maybe it should be called something like `DurationFromBeginning` to make it clear what it is?

Reply:
Would you find `ElapsedTime` a better name? This used to be a `time.Time`, hence the original naming, but then I realised it was better to perform the diff inside here rather than letting the person handling the events do it, as this model causes less pain to the user.

Reply:
Or maybe the problem is that `Duration` is unclear? It is the amount of time we have been blocked waiting for a blocking call to complete.

Reply:
How about `Duration` => `TimeWaited`?

Reply:
https://github.com/ooni/netx/issues/22